### PR TITLE
fix(server): Initialize ServerFamily with all listeners.

### PR DESCRIPTION
**Pull request by royjacobson**
- Add a test for CLIENT LIST which is the visible result of this.

### CODE REVIEW

File: src/server/main_service.cc [hunk 1]
```diff
@@ -557,7 +557,7 @@ void Service::Init(util::AcceptServer* acceptor, util::ListenerInterface* main_i
   request_latency_usec.Init(&pp_);
   StringFamily::Init(&pp_);
   GenericFamily::Init(&pp_);
-  server_family_.Init(acceptor, main_interface, &cluster_family_);
+  server_family_.Init(acceptor, listeners, &cluster_family_);
```
Comment (chakaz) [inline src/server/main_service.cc:560]: call with `std::move()`?

**Review by chakaz — APPROVED** [review]
Generally LGTM
However, since `AcceptServer` already has all listeners registered (and owned), and since `server_family_` already has a pointer to it, perhaps you could just expose them somehow and use that exposed list?

**Comment by royjacobson** [comment]
> Generally LGTM However, since `AcceptServer` already has all listeners registered (and owned), and since `server_family_` already has a pointer to it, perhaps you could just expose them somehow and use that exposed list?

I could, but I want the type to be `facade::Listener*` and not helio's `ListenerInterface`.

**Review by chakaz — APPROVED** [review]

File: src/server/dfly_main.cc [hunk 2]
```diff
@@ -381,9 +384,12 @@ bool RunEngine(ProactorPool* pool, AcceptServer* acceptor) {
       delete admin_listener;
     } else {
       LOG(INFO) << "Listening on " << printable_addr;
+      listeners.push_back(admin_listener);
```
Comment (romange) [inline src/server/dfly_main.cc:387]: I would rather not expose admin connections via "client list".

Comment (royjacobson) [reply src/server/dfly_main.cc:387]: We don't really have a well defined security boundary between and non-admin connections yet, and I'd rather not invent it on the fly... for example, should admin connections be hidden even when CLIENT LIST is executed from the admin connection?

Comment (romange) [reply src/server/dfly_main.cc:387]: Ok, opned https://github.com/dragonflydb/dragonfly/issues/1495 for that

**Pull request merged**